Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm

```diff
@@ -535,22 +535,51 @@ - (void)_didFinishLoadingDataForCustomContentProviderWithSuggestedFilename:(cons
     _page->didLayoutForCustomContentProvider();
 }
 
-ALLOW_DEPRECATED_IMPLEMENTATIONS_BEGIN
-- (void)_handleKeyUIEvent:(::UIEvent *)event
+static bool customContentViewDidHandleKeyEvent(UIPressesEvent *event, bool isFirstResponder, RetainPtr<UIView<WKWebViewContentProvider>> customContentView)
 {
     // We only want to handle key events from the hardware keyboard when we are
     // first responder and a custom content view is installed; otherwise,
     // WKContentView will be the first responder and expects to get key events directly.
-    if ([self isFirstResponder] && event._hidEvent) {
-        if ([_customContentView respondsToSelector:@selector(web_handleKeyEvent:)]) {
-            if ([_customContentView web_handleKeyEvent:event])
-                return;
+    if (isFirstResponder && event._hidEvent) {
+        if ([customContentView respondsToSelector:@selector(web_handleKeyEvent:)]) {
+            if ([customContentView web_handleKeyEvent:event])
+                return true;
         }
     }
+    return false;
+}
+
+- (void)pressesBegan:(NSSet<UIPress *> *)presses withEvent:(UIPressesEvent *)event
+{
+    if (customContentViewDidHandleKeyEvent(event, [self isFirstResponder], _customContentView))
```

Comment:
Nit - given that this helper function uses multiple states that are already on `WKWebView` (i.e. first responder status and `_customContentView`), I would just make this a helper method like
```- (BOOL)_tryToHandleKeyEventInCustomContentView:(UIPressesEvent *)event;```
...and then just use that in these three places.

Reply:
Changed this from a static function to a helper method